Created integrations for ad endpoints

autumnfound opened the pull request:
Signed-off-by: Martin Lowe <martin.lowe@eclipse-foundation.org>

autumnfound committed:
Created integrations for ad endpoints

Signed-off-by: Martin Lowe <martin.lowe@eclipse-foundation.org>

autumnfound committed:
Add pagination bar to all promos call

Signed-off-by: Martin Lowe <martin.lowe@eclipse-foundation.org>

ericpoirier committed:
Fetch the Ads from newsroom

Signed-off-by: Eric Poirier <eric.poirier@eclipse-foundation.org>

ericpoirier committed:
update dist files for conflicts

Signed-off-by: Eric Poirier <eric.poirier@eclipse-foundation.org>

autumnfound commented on src/jquery.eclipsefdn-api.js:
```diff
@@ -2039,38 +2041,38 @@
-        "ad_index": $container.data("index")
+        "ip": $container.data("ip-address")
       };
+      //var string_params = JSON.stringify(params);
```
We should remove this comment, but also should add the ability to fetch by ID so that we can inject a given ID on a page (we have this ability currently)

ericpoirier replied:
done

autumnfound commented on src/jquery.eclipsefdn-api.js:
```diff
@@ -1949,10 +1952,134 @@
+      var self = this;
+      var $container = $($(self)[0].element);
+      var url = self.settings.adsUrl;
+      var newsroomurl = self.settings.newsroomUrl;
```
What do we use this for? We should probably remove this

ericpoirier replied:
I use it on line 2070 to create the url of the default promo.

autumnfound commented on src/jquery.eclipsefdn-api.js:
```diff
@@ -1949,10 +1952,134 @@
+        filter += "&pagesize=" + displayCount;
+        filter += convertDataToURLParameters($parent, "publish-target", "publish_to", undefined);
+        
+        console.log(url + filter);
```
We should remove this console for final version

ericpoirier replied:
Done

autumnfound reviewed (commented):
I can't request changes on my own PRs, haha, but that's what I would have done :P

ericpoirier committed:
Remove console.log and remove dist files

Signed-off-by: Eric Poirier <eric.poirier@eclipse-foundation.org>

ericpoirier committed:
Fetch the Ads from newsroom

Conflicts:
	dist/jquery.eclipsefdn-api.js
	dist/jquery.eclipsefdn-api.min.js
	src/jquery.eclipsefdn-api.js

Signed-off-by: Eric Poirier <eric.poirier@eclipse-foundation.org>

ericpoirier committed:
Update dist file

Signed-off-by: Eric Poirier <eric.poirier@eclipse-foundation.org>

ericpoirier committed:
Add trigger for ads

Signed-off-by: Eric Poirier <eric.poirier@eclipse-foundation.org>

ericpoirier committed:
Remove unnecessary code from newsroom Ads

Signed-off-by: Eric Poirier <eric.poirier@eclipse-foundation.org>
